fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaActivity.java

```diff
@@ -152,46 +99,80 @@ public boolean onOptionsItemSelected(MenuItem item) {
     return super.onOptionsItemSelected(item);
   }
 
-  public void setupNavigationBar() {
-    BottomNavigationView bottomNavigationView = findViewById(R.id.social_media_nav_bar);
-    bottomNavigationView.setOnItemSelectedListener(
+  private void setupNavigationBar() {
+    navbar = findViewById(R.id.social_media_nav_bar);
+    navbar.setOnItemSelectedListener(
         item -> {
           int itemId = item.getItemId();
-          if (itemId == R.id.social_media_home_menu) {
-            mViewModel.openHome();
-          } else if (itemId == R.id.social_media_search_menu) {
-            mViewModel.openSearch();
-          } else if (itemId == R.id.social_media_following_menu) {
-            mViewModel.openFollowing();
-          } else if (itemId == R.id.social_media_profile_menu) {
-            mViewModel.openProfile();
+          if (itemId != mViewModel.getCurrentSelectedItem().getValue()) {
+            mViewModel.setCurrentSelectedItem(itemId);
+          } else {
+            if (itemId == R.id.social_media_home_menu) {
+              Log.d(TAG, "Opening home");
+              openSocialMediaHomeFragment();
+            } else if (itemId == R.id.social_media_search_menu) {
+              Log.d(TAG, "Opening search");
+              openSocialMediaSearchFragment();
+            } else if (itemId == R.id.social_media_following_menu) {
+              Log.d(TAG, "Opening following");
+              openSocialMediaFollowingFragment();
+            } else if (itemId == R.id.social_media_profile_menu) {
+              Log.d(TAG, "Opening profile");
+              openSocialMediaProfileFragment();
+            }
```

Comment:
same thing, this looks like it should be a switch. Is there a technical reason it can't be ? 🤔

Reply:
AS explained in the other comment, yes 😢